packages/webview_flutter/ios/Classes/FLTCookieManager.m

```diff
@@ -1,49 +1,69 @@
 #import "FLTCookieManager.h"
+#import "CookieDto.h"
 
 @implementation FLTCookieManager {
 }
 
+NSSet *websiteDataTypes;
+API_AVAILABLE(ios(9.0))
+WKWebsiteDataStore *dataStore;
+API_AVAILABLE(ios(11.0))
+WKHTTPCookieStore *cookieStore;
+
 + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   FLTCookieManager *instance = [[FLTCookieManager alloc] init];
 
   FlutterMethodChannel *channel =
       [FlutterMethodChannel methodChannelWithName:@"plugins.flutter.io/cookie_manager"
                                   binaryMessenger:[registrar messenger]];
   [registrar addMethodCallDelegate:instance channel:channel];
+
+  if (@available(iOS 11.0, *)) {
+    websiteDataTypes = [NSSet setWithArray:@[ WKWebsiteDataTypeCookies ]];
+    dataStore = [WKWebsiteDataStore defaultDataStore];
+    cookieStore = [dataStore httpCookieStore];
+  } else {
+    NSLog(@"This plugin is not supported for iOS versions prior to iOS 11.");
+  }
 }
 
-- (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
+- (void)handleMethodCall:(FlutterMethodCall *)call
+                  result:(FlutterResult)result API_AVAILABLE(ios(11.0)) {
   if ([[call method] isEqualToString:@"clearCookies"]) {
     [self clearCookies:result];
+  } else if ([[call method] isEqualToString:@"getCookies"]) {
+    [self getCookies:result];
+  } else if ([[call method] isEqualToString:@"setCookies"]) {
+    [self setCookies:call result:result];
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
-- (void)clearCookies:(FlutterResult)result {
-  if (@available(iOS 9.0, *)) {
-    [self clearCookiesIos9AndLater:result];
-  } else {
-    // support for iOS8 tracked in https://github.com/flutter/flutter/issues/27624.
-    NSLog(@"Clearing cookies is not supported for Flutter WebViews prior to iOS 9.");
+- (void)getCookies:(FlutterResult)result API_AVAILABLE(ios(11.0)) {
+  [cookieStore getAllCookies:^(NSArray<NSHTTPCookie *> *cookies) {
+    NSArray *serialized = [CookieDto manyToDictionary:[CookieDto manyFromNSHTTPCookies:cookies]];
+    result(serialized);
+  }];
+}
+
+- (void)setCookies:(FlutterMethodCall *)call result:(FlutterResult)result API_AVAILABLE(ios(11.0)) {
+  NSArray<CookieDto *> *cookieDtos = [CookieDto manyFromDictionaries:[call arguments]];
+  for (CookieDto *cookieDto in cookieDtos) {
+    [cookieStore setCookie:[cookieDto toNSHTTPCookie]
+         completionHandler:^(){
+         }];
```

Comment:
I tested this out and found several issues:

- the `result` handler will never be called => your async dart code is stuck, same for `clearCookies`
- the cookie data only considers `name` and `value`. So this is not going to work at all. In order for cookies to work we also need to consider other properties like the `domain`. But after adding the rest of the fields, and implementing some stupid counter code, I made it work.

Stupid counter code that calls `result` in the end example:

```objc
- (void)setCookies:(FlutterMethodCall *)call result:(FlutterResult)result API_AVAILABLE(ios(11.0)) {
    NSArray<CookieDto *> *cookieDtos = [CookieDto manyFromDictionaries:[call arguments]];
    __block int counter = 0;
    for (CookieDto *cookieDto in cookieDtos) {
        [cookieStore setCookie:[cookieDto toNSHTTPCookie]
             completionHandler:^(){
            counter++;
            if (counter == cookieDtos.count) {
                result(nil);
            }
        }];
    }
}
```

Reply:
Thanks. I'll use the code you provided until I or someone else comes up with a better solution.